aws/resource_aws_dms_replication_instance.go

```diff
@@ -286,9 +290,16 @@ func resourceAwsDmsReplicationInstanceUpdate(d *schema.ResourceData, meta interf
 		}
 	}
 
+	if d.HasChange("allow_major_version_upgrade") {
+		request.AllowMajorVersionUpgrade = aws.Bool(d.Get("allow_major_version_upgrade").(bool))
+		// Having allowing_major_version_upgrade by itself should not trigger ModifyReplicationInstance
+		// as it results in InvalidParameterCombination: No modifications were requested
+	}
```

Comment:
My suggestion here would be to just optionally include it, if it is enabled. Operators may just leave this enabled in development environments and this conditional would currently have no effect.

```suggestion
	if v, ok := d.GetOk("allow_major_version_upgrade"); ok {
		request.AllowMajorVersionUpgrade = aws.Bool(v.(bool))
		// Having allowing_major_version_upgrade by itself should not trigger ModifyReplicationInstance
		// as it results in InvalidParameterCombination: No modifications were requested
	}
```